src/components/Table/GridTable.stories.tsx

```diff
@@ -1136,3 +1136,99 @@ export function ToggleCustomCollapse() {
     </>
   );
 }
+
+type ExpandHeader = { id: "expandableHeader"; kind: "expandableHeader" };
+type Header = { id: "header"; kind: "header" };
+type ExpandableData = {
+  kind: "data";
+  data: {
+    firstName: string | undefined;
+    lastName: string | undefined;
+    birthdate: string | undefined;
+    age: number | undefined;
+    occupation: string | undefined;
+    manager: string | undefined;
+  };
+};
+type ExpandableRow = ExpandHeader | Header | ExpandableData;
+
+export function ExpandableColumns() {
+  const rows: GridDataRow<ExpandableRow>[] = useMemo(
+    () => [
+      // New reserved 'kind' "groupHeader" property for GridTable to position row correctly
+      { kind: "header", id: "header", data: {} },
+      { kind: "expandableHeader", id: "expandableHeader", data: {} },
+      {
+        kind: "data" as const,
+        id: `user:1`,
+        data: {
+          firstName: "Brandon",
+          lastName: "Dow",
+          birthdate: "Jan 29, 1986",
+          age: 36,
+          occupation: "Software Engineer",
+          manager: "Steve Thompson",
+        },
+      },
+    ],
+    [],
+  );
+
+  const columns: GridColumn<ExpandableRow>[] = useMemo(
+    () => [
+      selectColumn<ExpandableRow>({ sticky: "left" }),
+      column<ExpandableRow>({
+        expandableHeader: () => "Address",
+        header: emptyCell,
```

Comment:
comment:

I'm looking at the Story right now. I do find it a bit odd that only 1 of 4 columns is expandable, yet the `expandableHeader: ...` is what's providing the header for 75% of the coincidentally non-expandable table.

But I'm also curious how much of that is just implementation details. Can _I_ make a `{ kind: "header" }` that is also expandable, and a second row that is a "subheader" or something?

Reply:
That is an interesting take. Hopefully in practice there will be more than just one column that is expandable. Otherwise, seems like an unnecessary UX.
The idea of having a single row define multiple rows is an interesting take too. That might mean reworking some of our types, as we don't want any `GridDataRow` type to include multiple rows. I do think we can improve in this aspect, specifically now that we have 3 rows that should always be fixed in the table header. It'd be cleaner, as you suggest, to have a single:
```
const headerRow = { kind: "tableHead", id: "redundant... : )", data: {...} };

const someColumn = { 
  tableHead: {
    totals: (data) => ({ content: data.totalPrice }),
    expandableHeader: () => ({ content: "Expandable Title" }),
    header: () => ({ content: "Total price" }),
  },
  data: {...}
}
```
I think that is a fairly larger refactor in order to do this, but we could start it without impact the existing tables, which could be a cool way to move forward. Thanks for the suggestion and thoughts.